File: t5.1-TCP-ej/2-DayTimeClient.rb
require 'socket'    # importa libreria de sockets

host = (ARGV[0] || 'localhost')  # asignación condicional

begin

  sock=TCPSocket.new(host,2000)  # crea socket a “host” en puerto 13
  time = sock.gets             # lee el msj enviado por servidor
  puts "It is "+ time +" at "+host

  rescue Exception  # Excepción indica que no hay
    puts "no server at " + host # servidor activo
  ensure
    sock.close()    # ensure asegura ejecución haya excepcion o no 
end

File: t5.1-TCP-ej/2-DayTimeClient.java
import java.io.*;  
import java.net.*;

public class DayTime {  
  public static void main (String args[]) {   
   String host = args.length > 0 ? args[0]:"localhost";
   try {
      Socket s = new Socket(host, 13);
      BufferedReader reader = new BufferedReader(
            new InputStreamReader(s.getInputStream()));
      String time = reader.readLine();  
      s.close();
      System.out.println("It is " + time + " at " + host);
   } catch (UnknownHostException e) {
      System.err.println(e);
   } catch (IOException e) {
      System.err.println(e);
   } 
  }
}